File: 0442-Find-All-Duplicates-in-an-Array.js
var findDuplicates = function(nums) {
    let duplicates = [];

    for (let i = 0; i < nums.length; i++) {
        n = Math.abs(nums[i]);
        if (nums[n-1] < 0) {
            duplicates.push(n);
        }
        nums[n-1] *= -1;
    }

    return duplicates;
};

File: 0442-Find-All-Duplicates-in-an-Array.py
class Solution(object):
    def findDuplicates(self, nums):
        duplicates = []

        for i in range(len(nums)):
            n = abs(nums[i])
            if nums[n-1] < 0:
                duplicates.append(n)
            nums[n-1] *= -1

        return duplicates
